File: solutions/3346-lexicographically-smallest-string-after-operations-with-constraint/lexicographically-smallest-string-after-operations-with-constraint.java
class Solution {
    public String getSmallestString(String s, int k) {
        StringBuilder sb = new StringBuilder("");
        int n = s.length();
        for(int i = 0; i < n; i++){
            char c = s.charAt(i);
            if(c == 'a')
                sb.append('a');
            else{
                int min = Math.min(c - 'a', ('z' - c) + 1);
                if(min <= k){
                    k -= min;
                    sb.append('a');
                }
                else{
                    int y = ((c - 'a') - k);
                    sb.append((char)(y + 'a'));
                    k = 0;
                }
            }
        }
        return sb.toString();
    }
}

File: solutions/3346-lexicographically-smallest-string-after-operations-with-constraint/lexicographically-smallest-string-after-operations-with-constraint.cpp
class Solution {
public:
    string getSmallestString(string s, int k) {
        int n = s.size();
        string t = s;
        for(int i = 0; i < n; i++){
            int c = s.at(i)- 'a';
            for(int j = 0; j < 26; j++){
                int d = min(abs(c - j), 26 - abs(c - j));
                if(d <= k){
                    t.at(i) = char(j + 'a');
                    k -= d;
                    break;
                }
            }
        }
        return t;
    }
};